Microsoft.Azure.Cosmos/src/Handler/RequestMessage.cs

```
@@ -267,6 +267,13 @@ internal DocumentServiceRequest ToDocumentServiceRequest()
                     serviceRequest.UseGatewayMode = this.UseGatewayMode.Value;
                 }
 
+                if (this.RequestOptions?.Properties != null &&
+                   this.RequestOptions.Properties.TryGetValue("x-ms-cosmos-default-replica-index", out object defaultReplicaIndexHeader) &&
+                   int.TryParse(defaultReplicaIndexHeader.ToString(), out int defaultReplicaIndex))
```

Shouldn't it be considered an error if the value of the Property being sent is not a number instead of silently letting it pass? This would hide a potential bug on the calling layer that it's incorrectly setting some value and eventually failing the operation because the SDK did not route to a particular replica, and without any logs, it would be inferred that the SDK did a bad routing.

---

We can do the validation. Is this is a pattern to validate headers? Do we just need to log  the failures or throw a error ?

---

I believe it should throw the error so it is obvious and can be fixed. If just bypasses the header and logs a message it's very easy for developer to miss.